### ISSUE TITLE
"Stop" command should check state of minishift VM

### ISSUE THREAD

**Issue by agajdosi** [minishift/minishift#996]
We should improve around the way in which Minishift handles states of VM and commands which change these states. 

Great fact is that it is not possible to start a running VM anymore if the VM is running. However `minishift stop` command seem to be still not handling the situation correctly: 

```
[agajdosi@dhcp-24-156 ~]$ minishift status
Does Not Exist
[agajdosi@dhcp-24-156 ~]$ minishift stop
Stopping local OpenShift cluster...
Error stopping cluster:  Host does not exist: "minishift"
Error stopping cluster: Host does not exist: "minishift"
```

Similar situation is in case when VM is already stopped. I think it could be useful to prevent stop command from trying like it happens in `delete`:

```
[agajdosi@dhcp-24-156 ~]$ minishift delete
The execution of this command requires an existing 'minishift' VM. However, there is currently none defined.
```

**Comment by LalatenduMohanty** [comment]
@agajdosi Thanks. I agree that ` $ minishift stop` should check the state of VM and accordingly should provide the error.

**Comment by agajdosi** [comment]
We can also add a feature which will describe behaviour of Minishift when handling states with `start`, `stop` and `delete` commands. Focusing only on problematic cases it could look like this:

```cucumber
Feature: Minishift handles states of VM effectively
Minishift is able to change states of VM - can create, stop or delete a VM,
is able to report current state to a user and do not execute state changing commands
in command-state cases which are not supported - eg. starting a running VM. 

  Scenario: Deleting non existing VM
    Given Minishift has state "Does Not Exist"
     When executing "minishift delete"
     Then Minishift should have state "Does Not Exist"
      And stdout should contain
      """
      The execution of this command requires an existing 'minishift' VM. However, there is currently none defined.
      """
  Scenario: Stopping non existing VM
    Given Minishift has state "Does Not Exist"
     When executing "minishift stop"
     Then Minishift should have state "Does Not Exist"
      And stdout should contain
      """
      The execution of this command requires an existing 'minishift' VM. However, there is currently none defined.
      """

  Scenario: Stopping stopped VM
    Given Minishift has state "Stopped"
     When executing "minishift stop"
     Then Minishift should have state "Stopped"
      And stdout should contain
      """
      Minishift is already stopped.
      """

  Scenario: Starting running VM
    Given Minishift has state "Running"
     When executing "minishift start"
     Then Minishift should have state "Running"
      And stdout should contain
      """
      The 'minishift' VM is already running.
      """
```

**Comment by dharmit** [comment]
If I'm seeing this correctly, it's only the `minishift stop` command that's a concern here. Right?

**Comment by agajdosi** [comment]
@dharmit Yes, that is true.

**Comment by gbraad** [comment]
Can't assign an outsider as assignee... Will for now assign it to myself, but @dharmit will work on it

**Comment by hferentschik** [comment]
Merged via pull request #1010
